x-pack/auditbeat/module/system/sockets/sockets.go

```diff
@@ -0,0 +1,301 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux
+
+package sockets
+
+import (
+	"net"
+
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/auditbeat/cache"
+
+	"strconv"
+	"syscall"
+
+	"github.com/OneOfOne/xxhash"
+
+	"github.com/elastic/beats/libbeat/logp"
+	mbSocket "github.com/elastic/beats/metricbeat/module/system/socket"
+	"github.com/elastic/gosigar/sys/linux"
+)
+
+const (
+	moduleName    = "system"
+	metricsetName = "sockets"
```

Comment:
I think most of our metricsets use singular naming (e.g. process, core, filesystem).

Reply:
Done.